pkg/pusher/pusher.go

```diff
@@ -295,12 +295,15 @@ func (s *Service) checkReceipt(receipt *pushsync.Receipt) error {
 
 	po := swarm.Proximity(addr.Bytes(), peer.Bytes())
 	d := s.depther.NeighborhoodDepth()
+
+	// if the receipt po is out of depth AND the receipt has not yet hit the maximum retry limit, reject the receipt.
 	if po < d && s.attempts.try(addr) {
 		s.metrics.ShallowReceiptDepth.WithLabelValues(strconv.Itoa(int(po))).Inc()
 		return fmt.Errorf("pusher: shallow receipt depth %d, want at least %d", po, d)
 	}
 	s.logger.Tracef("pusher: pushed chunk %s to node %s, receipt depth %d", addr, peer, po)
 	s.metrics.ReceiptDepth.WithLabelValues(strconv.Itoa(int(po))).Inc()
+	s.attempts.delete(addr)
```

Comment:
since checking the receipt is not synchronized with setting the chunk as synced, this might result in the chunk being pushed again in the timeframe between the delete and the setting of the chunk. not critical but can result in a bit of redundant operations